### PR TITLE
Add realtime WPM in top right corner

### PR DESCRIPTION
**Pull request by himadriganguly**
This fixes enhancement #32

### CODE REVIEW

**Comment by Mithil467** [comment]
Hey @himadriganguly! Thanks for the PR.

Currently there are 2 issues with this feature.
- The wpm counter is only updating when a key is pressed, can you make it update every 1/10th of a second or something relative. Otherwise it seems very unpolished. But also pay attention you gotta use non-blocking input and no `while(true)` loops that will cause PCU usage 100% (see https://github.com/Mithil467/mitype/issues/12).
- Another small fix is the issue of cursor is moving to the end of the text as opposed to the last character you typed in sample text.

**Comment by Mithil467** [comment]
The latest source on master allows refreshing window every 100 milliseconds @himadriganguly :)

So realtime wpm is possible. Merge the latest code from master and you can continue working on this.
I did test it out locally and it does update every 100 ms but the wpm value is incorrect. Please look into this! 👍🏼

**Comment by Mithil467** [comment]
Hey @himadriganguly 👋🏼

Do comment if you are stuck somewhere or you may request review when you think it's required. 😄 

Some points to consider.

- The wpm value was refreshing earlier correctly, but the issue was with the wpm value. But now after the latest commit, it doesn't refresh every frame. 
- Current implementation of `speed_in_wpm` function in calculations.py considers text as the entire text sample. This was fine as we were wanting to calculate it only at the end earlier.
- But now that we want it every frame, an accurate implementation would be
```
current_wpm = (
60 * len(self.current_string.split()) / mitype.timer.get_elapsed_minutes_since_first_keypress(self.start_time)
)
```

This does give realtime wpm as expected :)

**Comment by himadriganguly** [comment]
@Mithil467 Have you checked the latest code that I have pushed? I haven't changed anything other than the WPM functionality that needs to be displayed. Anyways I will create another function for the WPM calculation and update the code.

**Comment by Mithil467** [comment]
Yes I did pull the latest changes. It doesn't update every frame but only upon key press. 

>I will create another function for the WPM calculation and update the code.

You may update the existing function, that would be better! 😄

File: mitype/app.py [hunk 1]
```diff
@@ -121,6 +121,9 @@ def typing_mode(self, win, key):
 
         self.key_strokes.append([time.time(), key])
 
+        if key:
```
Comment (Mithil467) [inline mitype/app.py:124]: Remove this if statement. Let the function be called even if no key is pressed, that way the stat is updated every frame.

**Review by Mithil467 — CHANGES_REQUESTED** [review]

**Comment by Mithil467** [comment]
@himadriganguly Thank you so much for this feature! This looks like a nice addition to the project. Good work. :)

**Pull request merged**